lib/device_interface.ts

```diff
@@ -80,20 +79,19 @@ export abstract class DeviceInterface {
 
   abstract async removeAllRawDatabasePayloads() : Promise<void>;
 
-  protected get keychainStorageKey() {
-    const keychainStorageKey = 'keychain';
-    if (this.namespace) {
-      return `${this.namespace}-${keychainStorageKey}`;
-    } else {
-      return keychainStorageKey;
-    }
-  }
+  abstract async getNamespacedKeychainValue() : Promise<any>;
+
+  abstract async setNamespacedKeychainValue(value: any) : Promise<void>;
 
-  abstract async getKeychainValue() : Promise<any>;
+  abstract async clearNamespacedKeychainValue() : Promise<void>;
 
-  abstract async setKeychainValue(value: any) : Promise<void>;
+  abstract async getRawKeychainValue() : Promise<any>;
 
-  abstract async clearKeychainValue() : Promise<void>;
+  abstract async clearRawKeychainValue() : Promise<void>;
+
+  public switchToNamespace(namespace: SNNamespace) {
```

Comment:
This should just be called `setNamespace`, since it is quite literally a setter. `switchTo` implies there's heavy gears spinning to make the switch, but really it's just a setter.